### PR TITLE
DRILL-7341: Vector reAlloc may fails after exchange

### PR DESCRIPTION
**Pull request by oleg-zinovev**
Fixes a relocate issues after direct  assignment of Vector data field.

### CODE REVIEW

File: exec/vector/src/main/codegen/templates/FixedValueVectors.java [hunk 1]
```diff
@@ -210,10 +210,16 @@ public void reAlloc() {
...
     // buffer if we start at 0.
 
-    final long newAllocationSize = allocationSizeInBytes == 0
+    long newAllocationSize = allocationSizeInBytes == 0
         ? 256
         : allocationSizeInBytes * 2L;
 
+    // Some operations, such as Value Vector#exchange, can be change DrillBuf data field without corresponding allocation size changes.
+    // Check that the size of the allocation is sufficient to copy the old buffer.
+    while (newAllocationSize < data.capacity()) {
```
Comment (weijietong) [inline exec/vector/src/main/codegen/templates/FixedValueVectors.java]: please also change the following 232 line: `data.setZero(halfNewCapacity, halfNewCapacity)` logic which initializes the allocated fresh buffer content to zero.

Comment (oleg-zinovev) [reply exec/vector/src/main/codegen/templates/FixedValueVectors.java]: Done.

Could you clarify why DrillBug#setZero performed only for FixedValueVectors?

Comment (weijietong) [reply exec/vector/src/main/codegen/templates/FixedValueVectors.java]: The change is to keep backward compatible. The reason I guess is that most of the fixed VV is number type. Initializing to zero is to keep the java default value convention.

**Comment by weijietong** [comment]
LGTM +1

**Pull request closed without merging**